db/versions/0091.yml

```diff
@@ -0,0 +1,391 @@
+  azure_queue_delete:
+    deprecated: true
+  azure_queue_update:
+    deprecated: true
+  azure_queue_put_extra:
+    deprecated: true
+  azure_queue_delete_expired:
+    deprecated: true
+
+  queue_pending_tasks_get:
+    description: |
+      Get up to `count` tasks for the pending tasks from the given taskQueueId.
+      Tasks are locked and will temporarily become invisible for the `visible` period.
+    mode: write
+    serviceName: queue
+    args: task_queue_id_in text, visible_in timestamptz, count integer
+    returns: table (task_id text, run_id integer, hint_id text, pop_receipt uuid)
+    body: |-
+      begin
+        return query
+          with updated as (
+            update queue_pending_tasks q
+            set pop_receipt = public.gen_random_uuid(),
+              visible = visible_in
+            where
+              q.task_id in (
+                select q2.task_id
+                from queue_pending_tasks q2
+                where q2.task_queue_id = task_queue_id_in
+                  and q2.visible <= now()
+                  and q2.expires > now()
+                order by q2.priority desc, q2.inserted asc
+                for update skip locked
+                limit count
+            )
+            returning q.priority, q.inserted, q.task_id, q.run_id, q.hint_id, q.pop_receipt
+          )
+          select u.task_id, u.run_id, u.hint_id, u.pop_receipt
+          from updated as u
+          order by u.priority desc, u.inserted asc;
+      end
+
+  queue_pending_tasks_put:
+    description: |
+      Put the task into the pending queue.
+      When record already exists, we update the priority, run_id, hint_id and expiration.
+      This also sends a notification to the `task_pending` channel with the `task_queue_id` as its payload.
+    mode: write
+    serviceName: queue
+    args: task_queue_id_in text, priority_in integer, task_id_in text, run_id_in integer, hint_id_in text, expires_in timestamp
+    returns: void
+    body: |-
+      begin
+        INSERT INTO queue_pending_tasks (task_queue_id, priority, task_id, run_id, hint_id, inserted, expires, visible)
+        VALUES (
+          task_queue_id_in,
+          priority_in,
+          task_id_in,
+          run_id_in,
+          hint_id_in,
+          now(),
+          expires_in,
+          now()
+        )
+        ON CONFLICT (task_id) DO UPDATE
+          SET
+            expires = greatest(coalesce(expires_in, queue_pending_tasks.expires), queue_pending_tasks.expires),
+            priority = priority_in,
+            run_id = greatest(run_id_in, queue_pending_tasks.run_id),
+            hint_id = hint_id_in
+          WHERE
+            queue_pending_tasks.task_queue_id = task_queue_id_in
+            AND queue_pending_tasks.task_id = task_id_in
+            AND queue_pending_tasks.pop_receipt is null;
+
+        -- notify listeners that there is a new task in the queue
+        EXECUTE 'NOTIFY task_pending, ' || quote_literal(task_queue_id_in) || ';';
```

Comment:
this will be used in followup PR. Listener would be added in `WorkClaimer` to listen to notifications and when `taskQueueId` in the payload will match listened one, claimer would fetch new tasks using existing functions.

I think that not sending full payload is fine, because we wouldn't be able to use that information directly anyway. And there might be zero listeners at the time. This is just to inform that new task is avaliable for the `taskQueueId`